src/frontend-samples/read-settings-sample/ReadSettingsWidget.tsx

```diff
@@ -0,0 +1,129 @@
+
+  const iModelConnection = useActiveIModelConnection();
+  const [settingKey, setSettingKey] = React.useState<string>(settingsKeys[0]);
+  const [settingValue, setSettingValue] = React.useState<string>();
+  const [settingResult, setSettingResult] = React.useState<SettingsResult>();
+  const [saveInProgress, setSaveInProgress] = React.useState<boolean>(false);
+  const [readUpdate, setReadUpdate] = React.useState<boolean>(true);
+  const [saveUpdate, setSaveUpdate] = React.useState<boolean>(false);
+
+  const _parseSettingsValue = useCallback((name: string, value: string) => {
+    let _value;
+    switch (name) {
+      case settingsKeys[0]:
+        _value = JSON.stringify(value || "", null, "   ");
+        break;
+      default:
+        _value = value || "";
+    }
+    return _value;
+  }, [settingsKeys]);
+
+  /** iModelConnection react state is very sensitive and updates a lot, so */
+  useEffect(() => {
+    if (iModelConnection && settingKey && readUpdate) {
+      ReadSettingsApi.readSettings(iModelConnection.iModelId!, iModelConnection.contextId!, settingKey).then((response) => {
+        setSettingResult(response);
+        setSettingValue(_parseSettingsValue(settingKey, response.setting));
+        setReadUpdate(false);
+      });
+    }
+  }, [_parseSettingsValue, iModelConnection, readUpdate, settingKey]);
+
+  // The showcase does not have permission to write data, it is expected to fail with 403 Forbidden.
+  useEffect(() => {
+    if (iModelConnection && saveInProgress && settingValue && saveUpdate) {
+      ReadSettingsApi.saveSettings(iModelConnection.iModelId!, iModelConnection.contextId!, settingKey, settingValue).then((response) => {
+        setSettingResult(response);
+        setSaveInProgress(false);
+        setSaveUpdate(false);
+      });
+    }
+  }, [iModelConnection, saveInProgress, saveUpdate, settingKey, settingValue]);
+
+  const _onSettingKeySelect = (event: ChangeEvent<HTMLSelectElement>) => {
+    setSettingKey(event.target.value);
+    setReadUpdate(true);
+  };
+
+  const _onSettingsValueChange = (event: any) => {
+    setSettingValue(event.target.value);
+    setSaveUpdate(true);
+  };
+
+  const _onSaveButtonPush = () => {
+    setSaveInProgress(true);
+    setSaveUpdate(true);
+  };
+
+  // Helper method to show status get/write operations with external setting in the dialog
+  const _showStatus = useCallback(() => {
+    if (!settingResult || settingResult.status === SettingsStatus.Success) {
+      return (<div></div>);
+    }
+
+    return (
+      <div style={{ lineBreak: "anywhere", overflowWrap: "break-word", maxWidth: "30vw" }}>
+        <SmallText style={{ color: "var(--foreground-alert)" }}>
+          {`${settingResult.status} ${settingResult.errorMessage}`}
+        </SmallText>
+      </div>
+    );
+  }, [settingResult]);
+
+  // Display drawing and sheet options in separate sections.
+  return (
+    <div className="sample-options">
+      {(!settingValue) && (
```

Comment:
Deleting all of a setting's value will result in a infinite spinner.

Repro:
- Select setting value textbox
- Ctrl + a
- Delete/backspace

Reply:
Checking on undefined rather than the object, because an empty string equates to false.